### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by Ezz1997**
Added course enrollment apis, kindly review it

### CODE REVIEW

File: models/userModel.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ const userSchema = new Schema(
 		email: { type: String, required: true, unique: true },
 		password: { type: String, required: true, minlength: 8 },
 		profilePhoto: { type: Buffer },
-		role: { type: String, enum: ['student', 'instructor', 'admin'], required: true },
+		role: { type: String, enum: ['student', 'instructor', 'admin'], required: false },
```
Comment (oti-adjei) [inline models/userModel.js]: why has it been set to false?

Comment (Ezz1997) [reply models/userModel.js]: in the signup routes the value isn't passed in the req and it's manually set as role: Role.Student or Role.Instructor, and as a result it caused an error, it solves the issue temporarily and forgot to change it, will revert it back and report the bug to Samad

Comment (oti-adjei) [reply models/userModel.js]: Noted

**Review by oti-adjei — APPROVED** [review]
Looks okay.I'll push mine after this.

**Pull request merged**